test/test_examples.py

```diff
@@ -147,16 +153,18 @@ def test_filename_regex(make_nc, filenames, expected):
 )
 def test_filename_regex_sorted(make_nc, filenames, expected):
 
-    wd = 'test/examples/ocean'
+    wd = Path('test/examples/ocean')
+    testfile = wd / 'test.nc'
 
     for filename in filenames:
-        filepath = f'{wd}/{filename}'
-        runcmd(f'cp {wd}/test.nc {filepath}')
+        filepath = wd / filename
+        os.makedirs(filepath.parent, exist_ok=True)
+        shutil.copy(testfile, filepath)
 
-    runcmd(r"addmeta -c addmetalist -v --sort --fnregex='^oceanbgc-\dd-(?P<variable>.*?)-(?P<frequency>.*?)-(?P<reduction>.*?)-??_\d+_\d+\.nc$'", wd)
+    runcmd(f"addmeta -c {wd}/addmetalist -v --sort --fnregex='oceanbgc-\dd-(?P<variable>.*?)-(?P<frequency>.*?)-(?P<reduction>.*?)-??_\d+_\d+\.nc$'")
```

Comment:
We can get rid of a warning from pytest with a raw string:
```suggestion
    runcmd(rf"addmeta -c {wd}/addmetalist -v --sort --fnregex='oceanbgc-\dd-(?P<variable>.*?)-(?P<frequency>.*?)-(?P<reduction>.*?)-??_\d+_\d+\.nc$'")
```